### PR TITLE
MAYA-112372: Add restriction for grouping.

### PR DESCRIPTION
**Pull request by HamedSabri-adsk**
group_blocking

### CODE REVIEW

File: lib/mayaUsd/ufe/UsdUndoCreateGroupCommand.cpp [hunk 1]
```diff
@@ -79,24 +79,36 @@ void UsdUndoCreateGroupCommand::execute()
...
+    } catch (...) {
+        // undo previous AddNewPrimCommand
+        undo();
 
-    TF_VERIFY(
-        Ufe::GlobalSelection::get()->size() == 1,
-        "_group node should be in the global selection now. \n");
+        throw; // re-throw the same exception
+    }
 }
```
Comment (HamedSabri-adsk) [inline lib/mayaUsd/ufe/UsdUndoCreateGroupCommand.cpp:113]: @seando-adsk 
Grouping in a two tier operation:

1- create a group node
2- insert selected item(s) under the group node.

There could be a case where creating a group node would be Ok by the restriction rules but not the insertion part.
For example, this scenario could happen if a user tries to group prim(s) in a layer other than the one where they were defined. In this case, the group creation itself will succeed but not the insertion part. In this case, we should throw an exception and undo the previous group command. 

Also keep in mind that the way we are performing the insertion here makes it impossible to support grouping flags (e.g absolute, relative, etc...) . Eventually, the insertion logics needs to be moved in Maya so we can support those flags.

**Comment by HamedSabri-adsk** [comment]
@seando-adsk for Maya's side of changes see MAYA-112372

File: test/lib/ufe/testGroupCmd.py [hunk 2]
```diff
@@ -241,6 +241,75 @@ def testGroupKind(self):
         self.assertEqual(Usd.ModelAPI(newGroupPrim).GetKind(), "")
         self.assertFalse(newGroupPrim.IsModel())
 
+    @unittest.skipUnless(mayaUtils.mayaMajorVersion() >= 2022, 'Grouping restriction is only available in Maya 2022 or greater.')
```
Comment (seando-adsk) [inline test/lib/ufe/testGroupCmd.py:244]: Does this really on Maya version, or is does it need UFE v3?

Comment (HamedSabri-adsk) [reply test/lib/ufe/testGroupCmd.py:244]: There is no UFE v3 required for changes in this PR.   The Maya side of changes which captures this exception is now in both master and release/2022 branch.

Comment (HamedSabri-adsk) [reply test/lib/ufe/testGroupCmd.py:244]: Also as a note, merge order doesn't really matter in the case of this PR.

**Review by seando-adsk — APPROVED** [review]

**Comment by HamedSabri-adsk** [comment]
@kxl-adsk  I am marking this PR as ready to merge. 

I ran the PF before and it all passed but one of the interactive fail which is not relevant here.

https://github.com/Autodesk/maya-usd/actions/runs/1006156994

**Pull request merged**